// @hyperledger/fabric-protos/lib/peer/policy_pb.js
// source: peer/policy.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var common_policies_pb = require('../common/policies_pb.js');
goog.object.extend(proto, common_policies_pb);
goog.exportSymbol('proto.protos.ApplicationPolicy', null, global);
goog.exportSymbol('proto.protos.ApplicationPolicy.TypeCase', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.protos.ApplicationPolicy = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.protos.ApplicationPolicy.oneofGroups_);
};
goog.inherits(proto.protos.ApplicationPolicy, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.protos.ApplicationPolicy.displayName = 'proto.protos.ApplicationPolicy';
}

/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.protos.ApplicationPolicy.oneofGroups_ = [[1,2]];

/**
 * @enum {number}
 */
proto.protos.ApplicationPolicy.TypeCase = {
  TYPE_NOT_SET: 0,
  SIGNATURE_POLICY: 1,
  CHANNEL_CONFIG_POLICY_REFERENCE: 2
};

/**
 * @return {proto.protos.ApplicationPolicy.TypeCase}
 */
proto.protos.ApplicationPolicy.prototype.getTypeCase = function() {
  return /** @type {proto.protos.ApplicationPolicy.TypeCase} */(jspb.Message.computeOneofCase(this, proto.protos.ApplicationPolicy.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.protos.ApplicationPolicy.prototype.toObject = function(opt_includeInstance) {
  return proto.protos.ApplicationPolicy.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.protos.ApplicationPolicy} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.protos.ApplicationPolicy.toObject = function(includeInstance, msg) {
  var f, obj = {
    signaturePolicy: (f = msg.getSignaturePolicy()) && common_policies_pb.SignaturePolicyEnvelope.toObject(includeInstance, f),
    channelConfigPolicyReference: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.protos.ApplicationPolicy}
 */
proto.protos.ApplicationPolicy.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.protos.ApplicationPolicy;
  return proto.protos.ApplicationPolicy.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.protos.ApplicationPolicy} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.protos.ApplicationPolicy}
 */
proto.protos.ApplicationPolicy.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new common_policies_pb.SignaturePolicyEnvelope;
      reader.readMessage(value,common_policies_pb.SignaturePolicyEnvelope.deserializeBinaryFromReader);
      msg.setSignaturePolicy(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setChannelConfigPolicyReference(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.protos.ApplicationPolicy.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.protos.ApplicationPolicy.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.protos.ApplicationPolicy} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.protos.ApplicationPolicy.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSignaturePolicy();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      common_policies_pb.SignaturePolicyEnvelope.serializeBinaryToWriter
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional common.SignaturePolicyEnvelope signature_policy = 1;
 * @return {?proto.common.SignaturePolicyEnvelope}
 */
proto.protos.ApplicationPolicy.prototype.getSignaturePolicy = function() {
  return /** @type{?proto.common.SignaturePolicyEnvelope} */ (
    jspb.Message.getWrapperField(this, common_policies_pb.SignaturePolicyEnvelope, 1));
};


/**
 * @param {?proto.common.SignaturePolicyEnvelope|undefined} value
 * @return {!proto.protos.ApplicationPolicy} returns this
*/
proto.protos.ApplicationPolicy.prototype.setSignaturePolicy = function(value) {
  return jspb.Message.setOneofWrapperField(this, 1, proto.protos.ApplicationPolicy.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.protos.ApplicationPolicy} returns this
 */
proto.protos.ApplicationPolicy.prototype.clearSignaturePolicy = function() {
  return this.setSignaturePolicy(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.protos.ApplicationPolicy.prototype.hasSignaturePolicy = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string channel_config_policy_reference = 2;
 * @return {string}
 */
proto.protos.ApplicationPolicy.prototype.getChannelConfigPolicyReference = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.protos.ApplicationPolicy} returns this
 */
proto.protos.ApplicationPolicy.prototype.setChannelConfigPolicyReference = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.protos.ApplicationPolicy.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.protos.ApplicationPolicy} returns this
 */
proto.protos.ApplicationPolicy.prototype.clearChannelConfigPolicyReference = function() {
  return jspb.Message.setOneofField(this, 2, proto.protos.ApplicationPolicy.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.protos.ApplicationPolicy.prototype.hasChannelConfigPolicyReference = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.protos);
