// @hyperledger/fabric-protos/lib/peer/signed_cc_dep_spec_pb.js
// source: peer/signed_cc_dep_spec.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var peer_proposal_response_pb = require('../peer/proposal_response_pb.js');
goog.object.extend(proto, peer_proposal_response_pb);
goog.exportSymbol('proto.protos.SignedChaincodeDeploymentSpec', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.protos.SignedChaincodeDeploymentSpec = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.protos.SignedChaincodeDeploymentSpec.repeatedFields_, null);
};
goog.inherits(proto.protos.SignedChaincodeDeploymentSpec, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.protos.SignedChaincodeDeploymentSpec.displayName = 'proto.protos.SignedChaincodeDeploymentSpec';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.protos.SignedChaincodeDeploymentSpec.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.toObject = function(opt_includeInstance) {
  return proto.protos.SignedChaincodeDeploymentSpec.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.protos.SignedChaincodeDeploymentSpec} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.protos.SignedChaincodeDeploymentSpec.toObject = function(includeInstance, msg) {
  var f, obj = {
    chaincodeDeploymentSpec: msg.getChaincodeDeploymentSpec_asB64(),
    instantiationPolicy: msg.getInstantiationPolicy_asB64(),
    ownerEndorsementsList: jspb.Message.toObjectList(msg.getOwnerEndorsementsList(),
    peer_proposal_response_pb.Endorsement.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.protos.SignedChaincodeDeploymentSpec}
 */
proto.protos.SignedChaincodeDeploymentSpec.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.protos.SignedChaincodeDeploymentSpec;
  return proto.protos.SignedChaincodeDeploymentSpec.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.protos.SignedChaincodeDeploymentSpec} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.protos.SignedChaincodeDeploymentSpec}
 */
proto.protos.SignedChaincodeDeploymentSpec.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setChaincodeDeploymentSpec(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setInstantiationPolicy(value);
      break;
    case 3:
      var value = new peer_proposal_response_pb.Endorsement;
      reader.readMessage(value,peer_proposal_response_pb.Endorsement.deserializeBinaryFromReader);
      msg.addOwnerEndorsements(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.protos.SignedChaincodeDeploymentSpec.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.protos.SignedChaincodeDeploymentSpec} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.protos.SignedChaincodeDeploymentSpec.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getChaincodeDeploymentSpec_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      1,
      f
    );
  }
  f = message.getInstantiationPolicy_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      2,
      f
    );
  }
  f = message.getOwnerEndorsementsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      peer_proposal_response_pb.Endorsement.serializeBinaryToWriter
    );
  }
};


/**
 * optional bytes chaincode_deployment_spec = 1;
 * @return {!(string|Uint8Array)}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getChaincodeDeploymentSpec = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * optional bytes chaincode_deployment_spec = 1;
 * This is a type-conversion wrapper around `getChaincodeDeploymentSpec()`
 * @return {string}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getChaincodeDeploymentSpec_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getChaincodeDeploymentSpec()));
};


/**
 * optional bytes chaincode_deployment_spec = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getChaincodeDeploymentSpec()`
 * @return {!Uint8Array}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getChaincodeDeploymentSpec_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getChaincodeDeploymentSpec()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.protos.SignedChaincodeDeploymentSpec} returns this
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.setChaincodeDeploymentSpec = function(value) {
  return jspb.Message.setProto3BytesField(this, 1, value);
};


/**
 * optional bytes instantiation_policy = 2;
 * @return {!(string|Uint8Array)}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getInstantiationPolicy = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes instantiation_policy = 2;
 * This is a type-conversion wrapper around `getInstantiationPolicy()`
 * @return {string}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getInstantiationPolicy_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getInstantiationPolicy()));
};


/**
 * optional bytes instantiation_policy = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getInstantiationPolicy()`
 * @return {!Uint8Array}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getInstantiationPolicy_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getInstantiationPolicy()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.protos.SignedChaincodeDeploymentSpec} returns this
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.setInstantiationPolicy = function(value) {
  return jspb.Message.setProto3BytesField(this, 2, value);
};


/**
 * repeated Endorsement owner_endorsements = 3;
 * @return {!Array<!proto.protos.Endorsement>}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.getOwnerEndorsementsList = function() {
  return /** @type{!Array<!proto.protos.Endorsement>} */ (
    jspb.Message.getRepeatedWrapperField(this, peer_proposal_response_pb.Endorsement, 3));
};


/**
 * @param {!Array<!proto.protos.Endorsement>} value
 * @return {!proto.protos.SignedChaincodeDeploymentSpec} returns this
*/
proto.protos.SignedChaincodeDeploymentSpec.prototype.setOwnerEndorsementsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.protos.Endorsement=} opt_value
 * @param {number=} opt_index
 * @return {!proto.protos.Endorsement}
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.addOwnerEndorsements = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.protos.Endorsement, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.protos.SignedChaincodeDeploymentSpec} returns this
 */
proto.protos.SignedChaincodeDeploymentSpec.prototype.clearOwnerEndorsementsList = function() {
  return this.setOwnerEndorsementsList([]);
};


goog.object.extend(exports, proto.protos);
