// @hyperledger/fabric-protos/lib/orderer/etcdraft/metadata_pb.js
// source: orderer/etcdraft/metadata.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

goog.exportSymbol('proto.etcdraft.BlockMetadata', null, global);
goog.exportSymbol('proto.etcdraft.ClusterMetadata', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.etcdraft.BlockMetadata = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.etcdraft.BlockMetadata.repeatedFields_, null);
};
goog.inherits(proto.etcdraft.BlockMetadata, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.etcdraft.BlockMetadata.displayName = 'proto.etcdraft.BlockMetadata';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.etcdraft.ClusterMetadata = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.etcdraft.ClusterMetadata.repeatedFields_, null);
};
goog.inherits(proto.etcdraft.ClusterMetadata, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.etcdraft.ClusterMetadata.displayName = 'proto.etcdraft.ClusterMetadata';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.etcdraft.BlockMetadata.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.etcdraft.BlockMetadata.prototype.toObject = function(opt_includeInstance) {
  return proto.etcdraft.BlockMetadata.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.etcdraft.BlockMetadata} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.etcdraft.BlockMetadata.toObject = function(includeInstance, msg) {
  var f, obj = {
    consenterIdsList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    nextConsenterId: jspb.Message.getFieldWithDefault(msg, 2, 0),
    raftIndex: jspb.Message.getFieldWithDefault(msg, 3, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.etcdraft.BlockMetadata}
 */
proto.etcdraft.BlockMetadata.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.etcdraft.BlockMetadata;
  return proto.etcdraft.BlockMetadata.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.etcdraft.BlockMetadata} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.etcdraft.BlockMetadata}
 */
proto.etcdraft.BlockMetadata.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var values = /** @type {!Array<number>} */ (reader.isDelimited() ? reader.readPackedUint64() : [reader.readUint64()]);
      for (var i = 0; i < values.length; i++) {
        msg.addConsenterIds(values[i]);
      }
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setNextConsenterId(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setRaftIndex(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.etcdraft.BlockMetadata.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.etcdraft.BlockMetadata.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.etcdraft.BlockMetadata} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.etcdraft.BlockMetadata.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getConsenterIdsList();
  if (f.length > 0) {
    writer.writePackedUint64(
      1,
      f
    );
  }
  f = message.getNextConsenterId();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getRaftIndex();
  if (f !== 0) {
    writer.writeUint64(
      3,
      f
    );
  }
};


/**
 * repeated uint64 consenter_ids = 1;
 * @return {!Array<number>}
 */
proto.etcdraft.BlockMetadata.prototype.getConsenterIdsList = function() {
  return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<number>} value
 * @return {!proto.etcdraft.BlockMetadata} returns this
 */
proto.etcdraft.BlockMetadata.prototype.setConsenterIdsList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.etcdraft.BlockMetadata} returns this
 */
proto.etcdraft.BlockMetadata.prototype.addConsenterIds = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.etcdraft.BlockMetadata} returns this
 */
proto.etcdraft.BlockMetadata.prototype.clearConsenterIdsList = function() {
  return this.setConsenterIdsList([]);
};


/**
 * optional uint64 next_consenter_id = 2;
 * @return {number}
 */
proto.etcdraft.BlockMetadata.prototype.getNextConsenterId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.etcdraft.BlockMetadata} returns this
 */
proto.etcdraft.BlockMetadata.prototype.setNextConsenterId = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional uint64 raft_index = 3;
 * @return {number}
 */
proto.etcdraft.BlockMetadata.prototype.getRaftIndex = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.etcdraft.BlockMetadata} returns this
 */
proto.etcdraft.BlockMetadata.prototype.setRaftIndex = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.etcdraft.ClusterMetadata.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.etcdraft.ClusterMetadata.prototype.toObject = function(opt_includeInstance) {
  return proto.etcdraft.ClusterMetadata.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.etcdraft.ClusterMetadata} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.etcdraft.ClusterMetadata.toObject = function(includeInstance, msg) {
  var f, obj = {
    activeNodesList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.etcdraft.ClusterMetadata}
 */
proto.etcdraft.ClusterMetadata.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.etcdraft.ClusterMetadata;
  return proto.etcdraft.ClusterMetadata.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.etcdraft.ClusterMetadata} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.etcdraft.ClusterMetadata}
 */
proto.etcdraft.ClusterMetadata.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var values = /** @type {!Array<number>} */ (reader.isDelimited() ? reader.readPackedUint64() : [reader.readUint64()]);
      for (var i = 0; i < values.length; i++) {
        msg.addActiveNodes(values[i]);
      }
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.etcdraft.ClusterMetadata.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.etcdraft.ClusterMetadata.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.etcdraft.ClusterMetadata} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.etcdraft.ClusterMetadata.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getActiveNodesList();
  if (f.length > 0) {
    writer.writePackedUint64(
      1,
      f
    );
  }
};


/**
 * repeated uint64 active_nodes = 1;
 * @return {!Array<number>}
 */
proto.etcdraft.ClusterMetadata.prototype.getActiveNodesList = function() {
  return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<number>} value
 * @return {!proto.etcdraft.ClusterMetadata} returns this
 */
proto.etcdraft.ClusterMetadata.prototype.setActiveNodesList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.etcdraft.ClusterMetadata} returns this
 */
proto.etcdraft.ClusterMetadata.prototype.addActiveNodes = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.etcdraft.ClusterMetadata} returns this
 */
proto.etcdraft.ClusterMetadata.prototype.clearActiveNodesList = function() {
  return this.setActiveNodesList([]);
};


goog.object.extend(exports, proto.etcdraft);
