// @hyperledger/fabric-protos/lib/peer/lifecycle/chaincode_definition_pb.js
// source: peer/lifecycle/chaincode_definition.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

goog.exportSymbol('proto.lifecycle.ChaincodeEndorsementInfo', null, global);
goog.exportSymbol('proto.lifecycle.ChaincodeValidationInfo', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.lifecycle.ChaincodeEndorsementInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.lifecycle.ChaincodeEndorsementInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.lifecycle.ChaincodeEndorsementInfo.displayName = 'proto.lifecycle.ChaincodeEndorsementInfo';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.lifecycle.ChaincodeValidationInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.lifecycle.ChaincodeValidationInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.lifecycle.ChaincodeValidationInfo.displayName = 'proto.lifecycle.ChaincodeValidationInfo';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.lifecycle.ChaincodeEndorsementInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.lifecycle.ChaincodeEndorsementInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.ChaincodeEndorsementInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    version: jspb.Message.getFieldWithDefault(msg, 1, ""),
    initRequired: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    endorsementPlugin: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.lifecycle.ChaincodeEndorsementInfo}
 */
proto.lifecycle.ChaincodeEndorsementInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.lifecycle.ChaincodeEndorsementInfo;
  return proto.lifecycle.ChaincodeEndorsementInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.lifecycle.ChaincodeEndorsementInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.lifecycle.ChaincodeEndorsementInfo}
 */
proto.lifecycle.ChaincodeEndorsementInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setVersion(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setInitRequired(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setEndorsementPlugin(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.lifecycle.ChaincodeEndorsementInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.lifecycle.ChaincodeEndorsementInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.ChaincodeEndorsementInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getVersion();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getInitRequired();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
  f = message.getEndorsementPlugin();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional string version = 1;
 * @return {string}
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.getVersion = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.lifecycle.ChaincodeEndorsementInfo} returns this
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.setVersion = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool init_required = 2;
 * @return {boolean}
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.getInitRequired = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.lifecycle.ChaincodeEndorsementInfo} returns this
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.setInitRequired = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};


/**
 * optional string endorsement_plugin = 3;
 * @return {string}
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.getEndorsementPlugin = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.lifecycle.ChaincodeEndorsementInfo} returns this
 */
proto.lifecycle.ChaincodeEndorsementInfo.prototype.setEndorsementPlugin = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.lifecycle.ChaincodeValidationInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.lifecycle.ChaincodeValidationInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.ChaincodeValidationInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    validationPlugin: jspb.Message.getFieldWithDefault(msg, 1, ""),
    validationParameter: msg.getValidationParameter_asB64()
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.lifecycle.ChaincodeValidationInfo}
 */
proto.lifecycle.ChaincodeValidationInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.lifecycle.ChaincodeValidationInfo;
  return proto.lifecycle.ChaincodeValidationInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.lifecycle.ChaincodeValidationInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.lifecycle.ChaincodeValidationInfo}
 */
proto.lifecycle.ChaincodeValidationInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setValidationPlugin(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setValidationParameter(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.lifecycle.ChaincodeValidationInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.lifecycle.ChaincodeValidationInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.ChaincodeValidationInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getValidationPlugin();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValidationParameter_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      2,
      f
    );
  }
};


/**
 * optional string validation_plugin = 1;
 * @return {string}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.getValidationPlugin = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.lifecycle.ChaincodeValidationInfo} returns this
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.setValidationPlugin = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bytes validation_parameter = 2;
 * @return {!(string|Uint8Array)}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.getValidationParameter = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes validation_parameter = 2;
 * This is a type-conversion wrapper around `getValidationParameter()`
 * @return {string}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.getValidationParameter_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getValidationParameter()));
};


/**
 * optional bytes validation_parameter = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getValidationParameter()`
 * @return {!Uint8Array}
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.getValidationParameter_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getValidationParameter()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.lifecycle.ChaincodeValidationInfo} returns this
 */
proto.lifecycle.ChaincodeValidationInfo.prototype.setValidationParameter = function(value) {
  return jspb.Message.setProto3BytesField(this, 2, value);
};


goog.object.extend(exports, proto.lifecycle);
