// @hyperledger/fabric-protos/lib/peer/lifecycle/db_pb.js
// source: peer/lifecycle/db.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

goog.exportSymbol('proto.lifecycle.StateData', null, global);
goog.exportSymbol('proto.lifecycle.StateData.TypeCase', null, global);
goog.exportSymbol('proto.lifecycle.StateMetadata', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.lifecycle.StateMetadata = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.lifecycle.StateMetadata.repeatedFields_, null);
};
goog.inherits(proto.lifecycle.StateMetadata, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.lifecycle.StateMetadata.displayName = 'proto.lifecycle.StateMetadata';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.lifecycle.StateData = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.lifecycle.StateData.oneofGroups_);
};
goog.inherits(proto.lifecycle.StateData, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.lifecycle.StateData.displayName = 'proto.lifecycle.StateData';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.lifecycle.StateMetadata.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.lifecycle.StateMetadata.prototype.toObject = function(opt_includeInstance) {
  return proto.lifecycle.StateMetadata.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.lifecycle.StateMetadata} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.StateMetadata.toObject = function(includeInstance, msg) {
  var f, obj = {
    datatype: jspb.Message.getFieldWithDefault(msg, 1, ""),
    fieldsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.lifecycle.StateMetadata}
 */
proto.lifecycle.StateMetadata.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.lifecycle.StateMetadata;
  return proto.lifecycle.StateMetadata.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.lifecycle.StateMetadata} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.lifecycle.StateMetadata}
 */
proto.lifecycle.StateMetadata.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDatatype(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addFields(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.lifecycle.StateMetadata.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.lifecycle.StateMetadata.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.lifecycle.StateMetadata} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.StateMetadata.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDatatype();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getFieldsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string datatype = 1;
 * @return {string}
 */
proto.lifecycle.StateMetadata.prototype.getDatatype = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.lifecycle.StateMetadata} returns this
 */
proto.lifecycle.StateMetadata.prototype.setDatatype = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string fields = 2;
 * @return {!Array<string>}
 */
proto.lifecycle.StateMetadata.prototype.getFieldsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.lifecycle.StateMetadata} returns this
 */
proto.lifecycle.StateMetadata.prototype.setFieldsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.lifecycle.StateMetadata} returns this
 */
proto.lifecycle.StateMetadata.prototype.addFields = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.lifecycle.StateMetadata} returns this
 */
proto.lifecycle.StateMetadata.prototype.clearFieldsList = function() {
  return this.setFieldsList([]);
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.lifecycle.StateData.oneofGroups_ = [[1,2,3]];

/**
 * @enum {number}
 */
proto.lifecycle.StateData.TypeCase = {
  TYPE_NOT_SET: 0,
  INT64: 1,
  BYTES: 2,
  STRING: 3
};

/**
 * @return {proto.lifecycle.StateData.TypeCase}
 */
proto.lifecycle.StateData.prototype.getTypeCase = function() {
  return /** @type {proto.lifecycle.StateData.TypeCase} */(jspb.Message.computeOneofCase(this, proto.lifecycle.StateData.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.lifecycle.StateData.prototype.toObject = function(opt_includeInstance) {
  return proto.lifecycle.StateData.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.lifecycle.StateData} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.StateData.toObject = function(includeInstance, msg) {
  var f, obj = {
    int64: jspb.Message.getFieldWithDefault(msg, 1, 0),
    bytes: msg.getBytes_asB64(),
    string: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.lifecycle.StateData}
 */
proto.lifecycle.StateData.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.lifecycle.StateData;
  return proto.lifecycle.StateData.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.lifecycle.StateData} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.lifecycle.StateData}
 */
proto.lifecycle.StateData.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setInt64(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setBytes(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setString(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.lifecycle.StateData.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.lifecycle.StateData.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.lifecycle.StateData} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.lifecycle.StateData.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {number} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = /** @type {!(string|Uint8Array)} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeBytes(
      2,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional int64 Int64 = 1;
 * @return {number}
 */
proto.lifecycle.StateData.prototype.getInt64 = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.setInt64 = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.lifecycle.StateData.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.clearInt64 = function() {
  return jspb.Message.setOneofField(this, 1, proto.lifecycle.StateData.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.lifecycle.StateData.prototype.hasInt64 = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional bytes Bytes = 2;
 * @return {!(string|Uint8Array)}
 */
proto.lifecycle.StateData.prototype.getBytes = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes Bytes = 2;
 * This is a type-conversion wrapper around `getBytes()`
 * @return {string}
 */
proto.lifecycle.StateData.prototype.getBytes_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getBytes()));
};


/**
 * optional bytes Bytes = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getBytes()`
 * @return {!Uint8Array}
 */
proto.lifecycle.StateData.prototype.getBytes_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getBytes()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.setBytes = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.lifecycle.StateData.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.clearBytes = function() {
  return jspb.Message.setOneofField(this, 2, proto.lifecycle.StateData.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.lifecycle.StateData.prototype.hasBytes = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string String = 3;
 * @return {string}
 */
proto.lifecycle.StateData.prototype.getString = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.setString = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.lifecycle.StateData.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.lifecycle.StateData} returns this
 */
proto.lifecycle.StateData.prototype.clearString = function() {
  return jspb.Message.setOneofField(this, 3, proto.lifecycle.StateData.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.lifecycle.StateData.prototype.hasString = function() {
  return jspb.Message.getField(this, 3) != null;
};


goog.object.extend(exports, proto.lifecycle);
