// @hyperledger/fabric-protos/lib/transientstore/transientstore_pb.js
// source: transientstore/transientstore.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var ledger_rwset_rwset_pb = require('../ledger/rwset/rwset_pb.js');
goog.object.extend(proto, ledger_rwset_rwset_pb);
var peer_collection_pb = require('../peer/collection_pb.js');
goog.object.extend(proto, peer_collection_pb);
goog.exportSymbol('proto.transientstore.TxPvtReadWriteSetWithConfigInfo', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.transientstore.TxPvtReadWriteSetWithConfigInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.transientstore.TxPvtReadWriteSetWithConfigInfo.displayName = 'proto.transientstore.TxPvtReadWriteSetWithConfigInfo';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.transientstore.TxPvtReadWriteSetWithConfigInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    endorsedAt: jspb.Message.getFieldWithDefault(msg, 1, 0),
    pvtRwset: (f = msg.getPvtRwset()) && ledger_rwset_rwset_pb.TxPvtReadWriteSet.toObject(includeInstance, f),
    collectionConfigsMap: (f = msg.getCollectionConfigsMap()) ? f.toObject(includeInstance, proto.protos.CollectionConfigPackage.toObject) : []
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.transientstore.TxPvtReadWriteSetWithConfigInfo;
  return proto.transientstore.TxPvtReadWriteSetWithConfigInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setEndorsedAt(value);
      break;
    case 2:
      var value = new ledger_rwset_rwset_pb.TxPvtReadWriteSet;
      reader.readMessage(value,ledger_rwset_rwset_pb.TxPvtReadWriteSet.deserializeBinaryFromReader);
      msg.setPvtRwset(value);
      break;
    case 3:
      var value = msg.getCollectionConfigsMap();
      reader.readMessage(value, function(message, reader) {
        jspb.Map.deserializeBinary(message, reader, jspb.BinaryReader.prototype.readString, jspb.BinaryReader.prototype.readMessage, proto.protos.CollectionConfigPackage.deserializeBinaryFromReader, "", new proto.protos.CollectionConfigPackage());
         });
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.transientstore.TxPvtReadWriteSetWithConfigInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getEndorsedAt();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getPvtRwset();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      ledger_rwset_rwset_pb.TxPvtReadWriteSet.serializeBinaryToWriter
    );
  }
  f = message.getCollectionConfigsMap(true);
  if (f && f.getLength() > 0) {
    f.serializeBinary(3, writer, jspb.BinaryWriter.prototype.writeString, jspb.BinaryWriter.prototype.writeMessage, proto.protos.CollectionConfigPackage.serializeBinaryToWriter);
  }
};


/**
 * optional uint64 endorsed_at = 1;
 * @return {number}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.getEndorsedAt = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} returns this
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.setEndorsedAt = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional rwset.TxPvtReadWriteSet pvt_rwset = 2;
 * @return {?proto.rwset.TxPvtReadWriteSet}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.getPvtRwset = function() {
  return /** @type{?proto.rwset.TxPvtReadWriteSet} */ (
    jspb.Message.getWrapperField(this, ledger_rwset_rwset_pb.TxPvtReadWriteSet, 2));
};


/**
 * @param {?proto.rwset.TxPvtReadWriteSet|undefined} value
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} returns this
*/
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.setPvtRwset = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} returns this
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.clearPvtRwset = function() {
  return this.setPvtRwset(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.hasPvtRwset = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * map<string, protos.CollectionConfigPackage> collection_configs = 3;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,!proto.protos.CollectionConfigPackage>}
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.getCollectionConfigsMap = function(opt_noLazyCreate) {
  return /** @type {!jspb.Map<string,!proto.protos.CollectionConfigPackage>} */ (
      jspb.Message.getMapField(this, 3, opt_noLazyCreate,
      proto.protos.CollectionConfigPackage));
};


/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.transientstore.TxPvtReadWriteSetWithConfigInfo} returns this
 */
proto.transientstore.TxPvtReadWriteSetWithConfigInfo.prototype.clearCollectionConfigsMap = function() {
  this.getCollectionConfigsMap().clear();
  return this;};


goog.object.extend(exports, proto.transientstore);
